scripts/AsmJsonImportTest.sh

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env bash
+
+set -e
+
+# Bash script to test the asm-json-import input mode of the compiler by
+# first exporting a .sol file to JSON that containing assembly json and corresponding bytecode,
+# then the compiler is invoked in assembly json import mode `--import-asm-json` and uses the previously
+# generated assembly json as input, where the corresponding bytecode output will be stored.
+# Finally, the originally generated bytecode will be compared with the one that was generated by using the
+# assembly json file as input.
+READLINK=readlink
+if [[ "$OSTYPE" == "darwin"* ]]; then
+    READLINK=greadlink
+fi
+REPO_ROOT=$(${READLINK} -f "$(dirname "$0")"/..)
+SOLIDITY_BUILD_DIR=${SOLIDITY_BUILD_DIR:-${REPO_ROOT}/build}
+SOLC=${SOLIDITY_BUILD_DIR}/solc/solc
+SPLITSOURCES=${REPO_ROOT}/scripts/splitSources.py
+
+SEMANTICTESTS_DIR="${REPO_ROOT}/test/libsolidity/semanticTests"
+NSOURCES="$(find "$SEMANTICTESTS_DIR" -type f | wc -l)"
+
+# DEV_DIR="${REPO_ROOT}/../tmp/contracts/"
+# NSOURCES="$(find $DEV_DIR -type f | wc -l)" #TODO use find command
+
+FAILED=0
+UNCOMPILABLE=0
+TESTED=0
+
+if [[ "$(find . -maxdepth 0 -type d -empty)" == "" ]]; then
+    echo "Test directory not empty. Skipping!"
+    exit 1
+fi
+
+# function tests whether importing an assembly json file creates identical bytecode.
+# Results are recorded by adding to FAILED or UNCOMPILABLE.
+# Also, in case of a mismatch a diff and the respective ASTs are printed
+# Expected parameters:
+# $1 name of the file to be exported and imported
+# $2 any files needed to do so that might be in parent directories
+function testImportExportEquivalence {
+    local nth_input_file="$1"
+    IFS=" " read -r -a all_input_files <<< "$2"
+
+    if $SOLC "$nth_input_file" "${all_input_files[@]}" > /dev/null 2>&1
+    then
+        # save exported json as expected result (silently)
+        $SOLC --combined-json asm,bin,compact-format --pretty-json "$nth_input_file" "${all_input_files[@]}" > expected.json 2> /dev/null
+        for contract in $(jq '.contracts | keys' expected.json | jq '.[]' 2> /dev/null)
```

Comment:
This adds a dependency on `jq`. We should list that in the docs about setting up dev environment and also ensure that we have it available in all images in CI. I guess we do since I do not see CI failing on Windows or macOS for this PR.

It's nice though. Now I'll be able to use it in other shell scripts :)